src/Microsoft.ComponentDetection/Program.cs

```diff
@@ -24,6 +24,7 @@
         .CreateBootstrapLogger();
 
     var serviceProvider = new ServiceCollection()
+        .AddHttpClient()
```

Comment:
This should be added in the `AddComponentDetection` method, as other code that uses CD as a library will fail to inject the `IHttpFactory`